mavros/src/plugins/sys_status.cpp

```diff
@@ -977,6 +1044,49 @@ class SystemStatusPlugin : public plugin::PluginBase
 		res.mode_sent = true;
 		return true;
 	}
+
+	bool vehicle_info_get_cb(mavros_msgs::VehicleInfoGet::Request &req,
+			mavros_msgs::VehicleInfoGet::Response &res)
+	{
+
+		if(req.get_all){
+			//Send all vehicles
+			for (auto &got : vehicles) {
+				res.vehicles.emplace_back(got.second);
```

Comment:
Perhaps here no difference between `push_back` and `emplace_back`.
Microoptimization - add `res.vehicles.reserve(vehicles.length())`.